output/json/json_test.go

```diff
@@ -47,7 +47,7 @@ func getValidator(t *testing.T, expected []string) func(io.Reader) {
 				t.Errorf("Read unexpected line number %d, expected only %d entries", i, len(expected))
 				continue
 			}
-			assert.JSONEq(t, expected[i-1], string(s.Bytes()))
+			assert.Equal(t, expected[i-1], string(s.Bytes()))
```

Comment:
`assert.JSONEq()` uses `json.Unmarshal()` and was hiding the encoding issue. So an equality check seems safer.

Reply:
Hmm but `Equal` may flakily fail if the order of encoding between the two differs, which it might when there are Go maps (e.g. `tags`) involved.

Reply:
go maps are sorted before they are encoded with the stdlib json. This is actually one of the "*optimizations*" in easyjson ;)

Reply:
Ah, that explains it then. I added another tag and [stress tested](https://dave.cheney.net/2013/06/19/stress-test-your-go-packages) `output/json` for a few minutes and didn't see a failure.

Besides, even if that wasn't the case we'd need to stabilize them for the tests, since `assert.JSONEq` is unusable because of the encoding issue.